hackathon_site/hardware/api_filters.py

```diff
@@ -0,0 +1,24 @@
+from django_filters import rest_framework as filters
+from .models import Hardware
+from .serializers import HardwareSerializer
+from django_filters.fields import Lookup
+
+
+class HardwareFilter(filters.FilterSet):
+    queryset = Hardware
+    serializer_class = HardwareSerializer
+
+    class Meta:
+        model = Hardware
+        fields = ["name", "quantity_available"]
+
+    def comma_separate_filter(self, queryset, value):
+        value_list = value.split(",")
+        queryset = super().filter(queryset, Lookup(value_list, "in"))
+
+        return queryset
+
+    in_stock = filters.BooleanFilter(
```

Comment:
On its own this doesn't quite work, since it's missing the "0" part of "> 0". This should work though:

```
    in_stock = filters.BooleanFilter(label="In stock", method="filter_in_stock")

    @staticmethod
    def filter_in_stock(queryset, _, value):
        if value is True:
            return queryset.filter(quantity_available__gt=0)
        else:
            return queryset.filter(quantity_available__lte=0)
```

You can read about filter methods [here](https://django-filter.readthedocs.io/en/stable/ref/filters.html#method). The methods take 3 parameters: `queryset`, `name` (the name of the field to filter on), and `value` (the value to use). 

For a `BooleanFilter`, the value under the hood will get converted to either `True` or `False`.  I'm ignoring the `name` parameter, since we can just hard code it to be `quantity_available`.

Slight problem for now though: we don't actually have `quantity_available` as a field yet, pending @YuYing-Liang's pr. For now, you can switch it to `quantity_available` for testing purposes.